src-built-in/components/shared/TagsMenu.jsx

```diff
@@ -0,0 +1,88 @@
+/*!
+* Copyright 2018 by ChartIQ, Inc.
+* All rights reserved.
+*
+*/
+
+import React from 'react'
+
+import './style.css';
+
+/**
+* A reusable flyout component for tags and any other lists
+* This component creates a button with a customizable label
+* and an item click handler.
+* Example: <TagsMenu label="Tags" align="left" list={list} onItemClick={handler}/>
+**/
+export default class TagsMenu extends React.Component {
+
+	constructor(props) {
+		super(props)
+		this.state = {
+			isVisible: false
+		}
+		// Bind context
+		this.toggleMenu = this.toggleMenu.bind(this)
+		this.onItemClick = this.onItemClick.bind(this)
+		this.setWrapperRef = this.setWrapperRef.bind(this);
+		this.handleClickOutside = this.handleClickOutside.bind(this);
+	}
+
+	componentDidMount() {
+		document.addEventListener('mousedown', this.handleClickOutside);
+	}
+
+	componentWillUnmount() {
+		document.removeEventListener('mousedown', this.handleClickOutside);
+	}
+
+	toggleMenu() {
+		this.setState({
+			isVisible: !this.state.isVisible
+		})
+	}
+
+	onItemClick(item) {
+		this.setState({
+			isVisible: false
+		})
+		this.props.onItemClick(item)
+	}
+
+	setWrapperRef(node) {
+		this.wrapperRef = node;
+	}
+
+	handleClickOutside(e) {
+		if (this.wrapperRef && !this.wrapperRef.contains(event.target)) {
+			this.setState({
+				isVisible: false
+			});
+		}
+	}
+
+	renderList() {
+		const items = this.props.list
+		const styles = this.props.align === 'right' ? { right: 0 } : { left: 0 }
```

Comment:
```suggestion
		const styles = {
		[this.props.align]: 0
		}
```

Putting `[` around a variable will interpolate it and create an object with 'right' or 'left'. If `this.props.align` will never be undefined, you're safe with this suggested syntax. Otherwise, what you have is better.